rancher-server/data.tf

```diff
@@ -47,11 +47,21 @@ data "helm_repository" "jetstack" {
   url  = "https://charts.jetstack.io"
 }
 
-data "template_file" "cloud_config" {
-  template = file("${path.module}/files/cloud-config.yaml")
-}
-
 data "rancher2_user" "admin" {
   username   = "admin"
   depends_on = [rancher2_bootstrap.admin]
 }
+
+data "aws_instances" "rancher_master" {
+  filter {
+    name   = "tag:aws:autoscaling:groupName"
+    values = [aws_autoscaling_group.rancher_master.0.name]
+  }
+}
+
+data "aws_instances" "rancher_worker" {
+  filter {
+    name   = "tag:aws:autoscaling:groupName"
+    values = [aws_autoscaling_group.rancher_worker.0.name]
+  }
+}
```

Comment:
Do these wait for the ASGs to reach capacity and for the instances to be ready before it returns?

Reply:
Yes, by default `aws_autoscaling_group` will wait until the ASG reaches desired capacity before returning, and us using `aws_autoscaling_group.rancher_worker.0.name` here makes an implied dependency in the graph